------------------------------------------------------------
revno: 748 [merge]
fixes bug: https://launchpad.net/bugs/1090205
author: Craig Tracey <craigtracey@gmail.com>
committer: Scott Moser <smoser@ubuntu.com>
branch nick: trunk
timestamp: Fri 2012-12-14 09:11:29 -0500
message:
  Provide a mechanism for puppet to be conditionally installed.
  
  Instead of always installing puppet give users the ability to opt-out.
------------------------------------------------------------
Use --include-merged or -n0 to see merged revisions.
=== modified file 'ChangeLog'

=== modified file 'cloudinit/config/cc_puppet.py'
--- a/cloudinit/config/cc_puppet.py
+++ b/cloudinit/config/cc_puppet.py
@@ -57,8 +57,10 @@ def handle(name, cfg, cloud, log, _args)
 
     puppet_cfg = cfg['puppet']
 
-    # Start by installing the puppet package ...
-    cloud.distro.install_packages(["puppet"])
+    # Start by installing the puppet package if necessary...
+    install = util.get_cfg_option_bool(puppet_cfg, 'install', True)
+    if install:
+        cloud.distro.install_packages(["puppet"])
 
     # ... and then update the puppet configuration
     if 'conf' in puppet_cfg:
